lua/format-on-save/init.lua

```diff
@@ -59,10 +59,13 @@ function M.setup(opts)
       end,
       group = augroup_id,
     })
-    vim.api.nvim_create_autocmd(
-      { "BufWritePost" },
-      { pattern = "*", callback = cursors.restore_current_buf_cursors, group = augroup_id }
-    )
+
+    if not opts.partial_update then
```

Comment:
can you change to `if opts.partial_update ~= "diff"` so it'll only be disabled for the new mode?

(I'm currently using the `partial_update = true` mode and I do want the cursor restoration to work for that)